packages/mobile/src/components/CurrencyPicker.tsx

```diff
@@ -0,0 +1,183 @@
+import Animated, { useAnimatedStyle, useSharedValue } from 'react-native-reanimated'
+import { SendEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import { useShowOrHideAnimation } from 'src/components/useShowOrHideAnimation'
+import { Namespaces } from 'src/i18n'
+import { useCurrencyBalance } from 'src/stableToken/hooks'
+import { CURRENCIES, Currency, CurrencyInfo, STABLE_CURRENCIES } from 'src/utils/currencies'
+
+export enum CurrencyPickerOrigin {
+  Send = 'Send',
+  Exchange = 'Exchange',
+}
+
+interface Props {
+  isVisible: boolean
+  origin: CurrencyPickerOrigin
+  onCurrencySelected: (currency: Currency) => void
+}
+
+function CurrencyOption({
+  currency,
+  currencyInfo,
+  onPress,
+}: {
+  currency: Currency
+  currencyInfo: CurrencyInfo
+  onPress: () => void
+}) {
+  const { t } = useTranslation(Namespaces.sendFlow7)
+  const balance = useCurrencyBalance(currency)
+  const amount = {
+    value: new BigNumber(balance ?? '0'),
+    currencyCode: currencyInfo.code,
+  }
+  return (
+    <TouchableOpacity
+      style={styles.currencyOptionContainer}
+      onPress={onPress}
+      testID={`${currencyInfo.code}Touchable`}
+    >
+      <Text style={styles.optionName}>{t('stableBalance', { token: currencyInfo.code })}</Text>
+      <View style={styles.currencyBalanceContainer}>
+        <CurrencyDisplay
+          style={styles.localBalance}
+          amount={amount}
+          showLocalAmount={true}
+          testID={`Local${currencyInfo.code}Balance`}
+        />
+        <CurrencyDisplay
+          style={styles.currencyBalance}
+          amount={amount}
+          showLocalAmount={false}
+          hideCode={false}
+          hideSymbol={true}
+          testID={`${currencyInfo.code}Balance`}
+        />
+      </View>
+    </TouchableOpacity>
+  )
+}
+
+const PICKER_HEIGHT = 250
+
+function CurrencyPicker({ isVisible, origin, onCurrencySelected }: Props) {
+  const [showingOptions, setOptionsVisible] = useState(isVisible)
+
+  const { t } = useTranslation(Namespaces.sendFlow7)
+
+  const onCurrencyPressed = (currency: Currency) => () => {
+    ValoraAnalytics.track(SendEvents.token_selected, {
+      origin,
+      token: CURRENCIES[currency].code,
+    })
+    onCurrencySelected(currency)
+  }
+
+  const progress = useSharedValue(0)
+  const animatedPickerPosition = useAnimatedStyle(() => ({
+    transform: [{ translateY: (1 - progress.value) * PICKER_HEIGHT }],
+  }))
+  const animatedOpacity = useAnimatedStyle(() => ({
+    opacity: 0.5 * progress.value,
+  }))
+
+  useShowOrHideAnimation(
+    progress,
+    isVisible,
+    () => setOptionsVisible(true),
+    () => setOptionsVisible(false)
+  )
+
+  if (!showingOptions) {
+    return null
+  }
+
+  return (
+    <View style={styles.container} testID="CurrencyPickerContainer">
+      <Animated.View style={[styles.background, animatedOpacity]} />
```

Comment:
Should we dismiss it when the background part is touched?

Reply:
Yes, do you mind if I do this in the next PR? Since this isn't hooked to anything yet it's a bit hard to test and it will save me time to add this in the refactor PR where I hook this component up with the SendAmount screen.